lib/coordinatorsharding.go

```diff
@@ -329,6 +353,28 @@ func (crd *Coordinator) PreprocessSharding(requests []*netstring.Netstring) (boo
 			crd.respond(ns.Serialized)
 			return true, ErrNoScuttleIdPredicate
 		}
+		//Capture ScuttleID column data in-case if it provided as part of query.
+		if (requests[i].Cmd == common.CmdBindName) && crd.isScuttleID(string(requests[i].Payload)) {
+			//To avoid repeated binds for scuttleId column
+			if !scuttleColumnPresent {
+				if i < (sz - 1) {
+					if requests[i+1].Cmd == common.CmdBindNum && requests[i+2].Cmd == common.CmdBindValueMaxSize {
+						scuttleID, _ = strconv.Atoi(string(requests[i+3].Payload))
+						scuttleColumnPresent = true
+					} else if requests[i+1].Cmd == common.CmdBindValue {
+						scuttleID, _ = strconv.Atoi(string(requests[i+1].Payload))
+						scuttleColumnPresent = true
+					} else {
+						if logger.GetLogger().V(logger.Verbose) {
+							logger.GetLogger().Log(logger.Verbose, crd.id, fmt.Sprintf("Bind value for scuttleID column: %s not present in Query.", GetConfig().ScuttleColName))
+						}
+						evt := cal.NewCalEvent(EvtTypeSharding, EvtNameBadScuttleId, cal.TransOK, fmt.Sprintf("Bind value for scuttleID column: %s not present in Query.", GetConfig().ScuttleColName))
+						evt.AddDataInt("sql", int64(uint32(crd.sqlhash)))
```

Comment:
I think scuttleColumnPresent still needs to set true and let verifyScuttleId returns error since the bind name exists.

Reply:
Fixed it